CMakeLists.txt

```diff
@@ -950,6 +951,26 @@ if (NOT BOOTSTRAPPING_MODE)
   message(FATAL_ERROR "turning off bootstrapping is not supported anymore")
 endif()
 
+option(SWIFT_MINIMAL_CXX_BOOTSTRAP
+    "Build a minimal Swift compiler from only C++ sources. This compiler is only suitable for bootstrapping a Swift toolchain in an environment where one isn't already available."
+    FALSE)
+
+if(SWIFT_MINIMAL_CXX_BOOTSTRAP)
+    message(WARNING "Building minimal Swift compiler from only C++ sources. This compiler is only suitable for bootstrapping a full Swift compiler.")
+
+  # Disable swift-syntax integration
+  set(SWIFT_BUILD_SWIFT_SYNTAX OFF)
+
+  # Disable regular expression support.
+  set(SWIFT_BUILD_REGEX_PARSER_IN_COMPILER OFF)
+
+  # Disable string processing library
+  set(SWIFT_ENABLE_EXPERIMENTAL_STRING_PROCESSING OFF)
```

Comment:
Foundation seems to need regex and string processing available to build. Of course, one can turn these back on with `extra-cmake-options` but that seems roundabout.

On the other hand, perhaps Foundation needs to change to avoid these features? That may require more care than is warranted when writing Foundation code, which may not be desirable.